test/jasmine/tests/parcoords_test.js

```diff
@@ -0,0 +1,889 @@
+        });
+
+        it('Calling `Plotly.restyle` with an object should amend the preexisting parcoords', function(done) {
+
+            var newStyle = Lib.extendDeep({}, mockCopy.data[0].line);
+            newStyle.colorscale = 'Viridis';
+            newStyle.reversescale = false;
+
+            Plotly.restyle(gd, {line: newStyle}).then(function() {
+
+                expect(gd.data.length).toEqual(1);
+
+                expect(gd.data[0].line.colorscale).toEqual('Viridis');
+                expect(gd.data[0].line.reversescale).toEqual(false);
+                expect(gd.data[0].dimensions.length).toEqual(11);
+                expect(gd.data[0].line.cmin).toEqual(-4000);
+                expect(gd.data[0].dimensions[0].constraintrange).toBeDefined();
+                expect(gd.data[0].dimensions[0].constraintrange).toEqual([100000, 150000]);
+                expect(gd.data[0].dimensions[1].constraintrange).not.toBeDefined();
+
+                done();
+            });
+
+
+        });
+
+        it('Should emit a \'plotly_restyle\' event', function(done) {
+
+            var tester = (function() {
+
+                var eventCalled = false;
+
+                return {
+                    set: function(d) {eventCalled = d;},
+                    get: function() {return eventCalled;}
+                };
+            })();
+
+            gd.on('plotly_restyle', function() {
+                tester.set(true);
+            });
+
+            expect(tester.get()).toBe(false);
+            Plotly.restyle(gd, 'line.colorscale', 'Viridis')
+                .then(window.setTimeout(function() {
+                    expect(tester.get()).toBe(true);
+                    done();
+                }, 0));
+
+        });
+
+        it('Should emit a \'plotly_hover\' event', function(done) {
+
+            var tester = (function() {
+
+                var eventCalled = false;
+
+                return {
+                    set: function(d) {eventCalled = d;},
+                    get: function() {return eventCalled;}
+                };
+            })();
+
+            gd.on('plotly_hover', function(d) {
+                tester.set({hover: d});
+            });
+
+            gd.on('plotly_unhover', function(d) {
+                tester.set({unhover: d});
+            });
+
+            expect(tester.get()).toBe(false);
+
+            mouseEvent('mousemove', 111, 207);
+            mouseEvent('mouseover', 111, 207);
+
+            window.setTimeout(function() {
+
+                expect(tester.get().hover.curveNumber).toBe(36);
+
+                mouseEvent('mousemove', 110, 126);
+                mouseEvent('mouseover', 110, 126);
+
+                window.setTimeout(function() {
+
+                    expect(tester.get().unhover.curveNumber).toBe(null);
+                    done();
+                }, 0);
+
+            }, 0);
+
+        });
+
+        it('Calling `Plotly.relayout` with string should amend the preexisting parcoords', function(done) {
```

Comment:
Fixed by the changes you proposed: https://github.com/plotly/plotly.js/pull/1256/commits/3ec4381438e16ab0c35541b70d5a5325b1ac79b5